ndb/src/google/cloud/ndb/model.py

```diff
@@ -439,6 +451,204 @@ def _verify_validator(validator):
 
         return validator
 
+    def __repr__(self):
+        """Return a compact unambiguous string representation of a property.
+
+        This cycles through all stored attributes and displays the ones that
+        differ from the default values.
+        """
+        args = []
+        cls = self.__class__
+        combined_attributes = (
+            cls._positional_attributes + cls._keyword_attributes
+        )
+        for index, attr in enumerate(combined_attributes):
+            instance_val = getattr(self, attr)
+            default_val = getattr(cls, attr)
+            if instance_val is not default_val:
+                if isinstance(instance_val, type):
+                    as_str = instance_val.__name__
+                else:
+                    as_str = repr(instance_val)
+
+                if index >= len(cls._positional_attributes):
+                    if attr.startswith("_"):
+                        attr = attr[1:]
+                    as_str = "{}={}".format(attr, as_str)
+                args.append(as_str)
+
+        return "{}({})".format(self.__class__.__name__, ", ".join(args))
+
+    def _datastore_type(self, value):
+        """Internal hook used by property filters.
+
+        Sometimes the low-level query interface needs a specific data type
+        in order for the right filter to be constructed. See
+        :meth:`_comparison`.
+
+        Args:
+            value (Any): The value to be converted to a low-level type.
+
+        Returns:
+            Any: The passed-in ``value``, always. Subclasses may alter this
+            behavior.
+        """
+        return value
+
+    def _comparison(self, op, value):
+        """Internal helper for comparison operators.
+
+        Args:
+            op (str): The comparison operator. One of ``=``, ``!=``, ``<``,
+                ``<=``, ``>``, ``>=`` or ``in``.
+
+        Returns:
+            FilterNode: A FilterNode instance representing the requested
+            comparison.
+
+        Raises:
+            BadFilterError: If the current property is not indexed.
+        """
+        # Import late to avoid circular imports.
+        from google.cloud.ndb import query
+
+        if not self._indexed:
+            raise exceptions.BadFilterError(
+                "Cannot query for unindexed property {}".format(self._name)
+            )
+
+        if value is not None:
+            value = self._datastore_type(value)
```

Comment:
There are actually three checks / conversions done here but I left out the other two to make this PR easier to review.

Reply:
They get added in #6295